src/frontend/src/binder/expr/function.rs

```diff
@@ -53,8 +55,64 @@ pub const SYS_FUNCTION_WITHOUT_ARGS: &[&str] = &[
     "current_timestamp",
 ];
 
+/// Evaluate the value of expression from
+/// sql user defined function
+/// NOTE: the current implementation is very simple and
+/// acts as a prototype to illustrate the overall logic
+fn sql_udf_expr_evaluate(
+    query: Query,
+    param_index_mapping: &HashMap<String, usize>,
+    input: &Vec<FunctionArg>,
+) -> Result<ExprImpl> {
+    let SetExpr::Select(select) = query.body else {
+        panic!("Invalid syntax");
+    };
+    let projection = select.projection;
+    let SelectItem::UnnamedExpr(expr) = projection
+        .into_iter()
+        .exactly_one()
+        .expect("`projection` should contain only one `SelectItem`")
+    else {
+        unreachable!("`projection` should contain only one `SelectItem`");
+    };
+    println!("Current expr: {:#?}", expr);
+
+    /// Currently only support i32, just as a simple prototype
+    fn expr_inner(
+        expression: ast_expr,
+        map: &HashMap<String, usize>,
+        input: &Vec<FunctionArg>,
+    ) -> i32 {
```

Comment:
We should not evaluate the function but inline it. Just bind the expression, replace input arguments during the binding, and return an `ExprImpl`. Some modifications to the binder may be required to let it know it is binding a function definition instead of a query.